language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToConfigurableCodeAction.java

```diff
@@ -0,0 +1,210 @@
+import io.ballerina.tools.diagnostics.Location;
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.codeaction.CodeActionNodeValidator;
+import org.ballerinalang.langserver.codeaction.CodeActionUtil;
+import org.ballerinalang.langserver.common.constants.CommandConstants;
+import org.ballerinalang.langserver.common.utils.PositionUtil;
+import org.ballerinalang.langserver.commons.CodeActionContext;
+import org.ballerinalang.langserver.commons.codeaction.spi.RangeBasedCodeActionProvider;
+import org.ballerinalang.langserver.commons.codeaction.spi.RangeBasedPositionDetails;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Code Action to convert module variables to configurable variable.
+ *
+ * @since 2201.10.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class ConvertToConfigurableCodeAction implements RangeBasedCodeActionProvider {
+
+    private static final String NAME = "Convert to configurable";
+    private static final String CONFIGURABLE = "configurable";
+    private static final List<String> LIST_UPDATING_METHODS = List.of("push", "pop", "remove", "removeAll", "reverse",
+            "setLength", "shift", "slice", "unshift");
+    private static final List<String> TABLE_UPDATING_METHODS = List.of("add", "put", "remove", "removeAll",
+            "removeIfHasKey");
+    private static final List<String> MAPPING_UPDATING_METHODS = List.of("remove", "removeAll", "removeIfHasKey");
+
+    @Override
+    public List<SyntaxKind> getSyntaxKinds() {
+        return List.of(SyntaxKind.BOOLEAN_LITERAL, SyntaxKind.NUMERIC_LITERAL, SyntaxKind.STRING_LITERAL,
+                SyntaxKind.UNARY_EXPRESSION, SyntaxKind.XML_TEMPLATE_EXPRESSION, SyntaxKind.STRING_TEMPLATE_EXPRESSION,
+                SyntaxKind.MAPPING_CONSTRUCTOR, SyntaxKind.LIST_CONSTRUCTOR, SyntaxKind.TABLE_CONSTRUCTOR);
+    }
+
+    @Override
+    public boolean validate(CodeActionContext context, RangeBasedPositionDetails positionDetails) {
+        NonTerminalNode matchedNode = positionDetails.matchedCodeActionNode();
+        NonTerminalNode parentNode = matchedNode.parent();
+        if (parentNode.kind() != SyntaxKind.MODULE_VAR_DECL
+                || !CodeActionNodeValidator.validate(context.nodeAtRange())) {
+            return false;
+        }
+        ModuleVariableDeclarationNode modVarDeclarationNode = (ModuleVariableDeclarationNode) parentNode;
+        if (hasIsolatedOrFinalOrConfigurableQualifier(modVarDeclarationNode)) {
+            return false;
+        }
+        SemanticModel semanticModel = context.currentSemanticModel().get();
+        Optional<Symbol> symbol = semanticModel.symbol(modVarDeclarationNode);
+        if (symbol.isEmpty() || symbol.get().kind() != SymbolKind.VARIABLE) {
+            return false;
```

Comment:
We can add a negative test from LS to cover this line.

We can do something to validate an invalid position.

Reply:
I am not sure we can do this. This was to added to make this safe. Can you provide a sample if you have in mind?

Reply:
How about (errored) code without a type? This is a module-variable without a proper kind.

```
PORT = 8080;
```